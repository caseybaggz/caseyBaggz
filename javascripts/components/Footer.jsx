import React from 'react';
import { NavItems } from '../NavItems';

export default class Footer extends React.Component {
  constructor() {
    super();
    this.items = NavItems;
  }

  render() {
    return (
      <footer className="footer-container">
        <ul className="social-list">
          {
            this.items.map((item, index) => {
              return (
                <li className={ 'item ' + item.name } key={ index }>
                  <a href={ item.link } className="link" target="_blank" />
                </li>
              );
            })
          }
        </ul>
      </footer>
    );
  }
}
