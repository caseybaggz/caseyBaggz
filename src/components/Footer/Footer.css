.Footer {
  padding: 1em 0;
}

.Footer-social-list {
  align-items: center;
  display: flex;
  justify-content: center;
}
