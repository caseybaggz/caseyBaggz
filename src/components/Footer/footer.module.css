.footer {
  align-items: center;
  color: #BBDEFB;
  display: flex;
  justify-content: center;
  padding-bottom: 24px;
  padding-top: 24px;
  width: 100%;
}