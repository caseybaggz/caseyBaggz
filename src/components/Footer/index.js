// @flow

import React, { Component } from 'react';

import FooterStyles from './footer.module.css';

const footer = (
  <footer className={FooterStyles.footer}>
    copyright Casey Baggz 2018 / #keepitminimalbruh
  </footer>
);

export default class Footer extends Component<{},{}> {
  shouldComponentUpdate(): boolean {
    return false;
  }

  render() {
    return footer;
  }
}