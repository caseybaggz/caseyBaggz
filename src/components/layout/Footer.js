// @flow

import React from 'react';
import styled from 'styled-components';
import FooterButton from '../actions/FooterButton';

const Wrapper = styled.footer`
  padding-bottom: 25px;
  padding-top: 75px;
`;

const FooterLinks = styled.ul`
  align-items: center;
  display: flex;
  padding-left: 47px;
  width: 100%;

  li {
    margin-right: 22px;
  }
`;

const footerContent: React$Node = (
  <Wrapper>
    <FooterLinks>
      <li>
        <FooterButton href="https://twitter.com/caseybaggz">
          twitter
        </FooterButton>
      </li>
    </FooterLinks>
  </Wrapper>
);

type Props = {};

function Footer(props: Props): React$Node {
  return footerContent;
}

export default React.memo(Footer);