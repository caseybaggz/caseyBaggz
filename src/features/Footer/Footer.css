.Footer {
  padding-bottom: 2rem;
  width: 100%;
}

.Footer-ul {
  align-items: center;
  display: flex;
  justify-content: center;
  list-style: none;
  width: 100%;
}

.Footer-li {
  display: inline-block;
  height: 3.125rem;
  margin-left: 1rem;
  width: 3.125rem;
}

.Footer-li:first-of-type {
  margin-left: 0;
}

.Footer-link {
  background-color: #556574;
  border-radius: 50%;
  display: inline-block;
  height: 100%;
  padding: 0.7rem;
  transition: background-color 150ms ease-out;
  width: 100%;
}

.Footer-link:hover {
  background-color: #202020;
}
