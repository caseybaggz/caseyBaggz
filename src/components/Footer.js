import 'whatwg-fetch';
import React, { Component } from 'react';
import FooterLink from './FooterLink';
import FetchHelper from '../classes/FetchHelper';
import '../stylesheets/Footer.css';


export default class Footer extends Component {
  constructor() {
    super();

    this.state = {
      items: []
    };
  }

  componentDidMount() {
    fetch('https://raw.githubusercontent.com/caseybaggz/caseybaggz/master/api/social-links.json')
      .then(FetchHelper.status)
      .then(FetchHelper.parseJSON)
      .then((data) => {
        this.setState({ items: data });
      })
      .catch((e) => {
        console.warn(e);
      });
  }

  render() {
    return (
      <footer className="footer-container">
        <ul className="social-list">
          {
            this.state.items.map((item, index) => (
              <FooterLink { ...item } key={ index } />
            ))
          }
        </ul>
      </footer>
    );
  }
}
