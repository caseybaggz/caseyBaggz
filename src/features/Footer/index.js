import React, { Component } from 'react';

import { Image } from 'features';

import {
  Github,
  LinkedIn,
  Twitter,
  s6
} from 'images';

import './Footer.css';


export default class Footer extends Component {
  constructor() {
    super();
    this.links = [
      { id: 0, img: Github, url: 'https://github.com/caseybaggz' },
      { id: 1, img: LinkedIn, url: 'https://www.linkedin.com/in/caseybaggz/' },
      { id: 2, img: Twitter, url: 'https://twitter.com/caseybaggz' },
      { id: 3, img: s6, url: 'https://society6.com/zart' }
    ];
  }

  getItems = () => {
    return this.links.map((link, eq) => (
      <li
        className="Footer-li"
        key={link.id}
      >
        <a
          className="Footer-link"
          href={link.url}
          target="_blank"
          rel="noopener noreferrer"
        >
          <Image
            alt="social icon"
            src={link.img}
          />
        </a>
      </li>
    ));
  }

  render() {
    const items = this.getItems();

    return(
      <div className="Footer">
        <ul className="Footer-ul">
          {items}
        </ul>
      </div>
    );
  }
}
