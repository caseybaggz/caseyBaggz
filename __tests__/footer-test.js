import React from 'react';
import Footer from '../javascripts/components/Footer';
import renderer from 'react-test-renderer';


it('should render Footer component successfully', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
