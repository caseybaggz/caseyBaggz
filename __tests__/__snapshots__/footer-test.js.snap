exports[`test should render Footer component successfully 1`] = `
<footer
  className="footer-container">
  <ul
    className="social-list">
    <li
      className="item github">
      <a
        className="link"
        href="https://github.com/caseybaggz"
        target="_blank" />
    </li>
    <li
      className="item linkedin">
      <a
        className="link"
        href="https://www.linkedin.com/in/caseybaggz"
        target="_blank" />
    </li>
    <li
      className="item twitter">
      <a
        className="link"
        href="https://twitter.com/caseybaggz"
        target="_blank" />
    </li>
    <li
      className="item codepen">
      <a
        className="link"
        href="http://codepen.io/caseybaggz/"
        target="_blank" />
    </li>
  </ul>
</footer>
`;
